### ISSUE TITLE
Custom AsterNET Exceptions

### ISSUE THREAD

**Issue by Deantwo** [AsterNET/AsterNET#199]
Better exception handling with custom exceptions.

Current examples:
- A `SystemException` is thrown when the AMI connection isn't up at the time of sending a command.
- An `AuthenticationFailedException` is thrown when the AMI connection fails a login attempt.
- And `TimeoutException` also seem to be used many places.

This seems weird and would likely be better to make some custom exceptions for the AsterNET.

I would suggest a new root exception be added for all AsterNET exceptions, and than make child exceptions from that.
For example:
- AsternetException : Exception
- AsternetAuthenticationFailedException : AsternetException
- AsternetTimeoutException : AsternetException
- AsternetConnectionException : AsternetException
- ...

See: #161

_Originally posted by @Deantwo in https://github.com/AsterNET/AsterNET/issues/29#issuecomment-445832173_

**Comment by skrusty** [comment]
Agreed. Better exceptions will allow for better handling of issues. Having a base impementation which can be caught, then handled out based on the type makes a lot more sense, and cleaner code. Also it limits the use of random exceptions here and there. Any uncaught exceptions within the framework can be spotted and encapsulated.
